### PR TITLE
chore(TypeScript): avoid ts-ignore

### PR DESCRIPTION
**Pull request by Haroenv**
- addMethods with `as`
- updateApiKey with type guard
- browseRules & browseSynonyms with `as any` (the key exists, but isn't supposed to be shown to the user, thus isn't in the type

### CODE REVIEW

File: packages/client-search/src/methods/client/updateApiKey.ts [hunk 1]
```diff
@@ -35,17 +35,16 @@ export const updateApiKey = (base: SearchClient) => {
       'description',
       'maxQueriesPerIPPerHour',
       'maxHitsPerQuery',
-    ];
+    ] as const;
```
Comment (eunjae-lee) [inline packages/client-search/src/methods/client/updateApiKey.ts:38]: just curiosity, there's no mutation on `apiKeyFields` below, and why `as const`?

Comment (Haroenv) [reply packages/client-search/src/methods/client/updateApiKey.ts:38]: without `as const`, the type of `updatedField` lower is string, not the actual field values

Comment (eunjae-lee) [reply packages/client-search/src/methods/client/updateApiKey.ts:38]: Ah I didn't know. Thanks!

**Review by eunjae-lee — APPROVED** [review]
This looks good to me.

**Pull request merged**